src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -71,6 +72,8 @@ void IListener.Cancel()
 
         void IDisposable.Dispose()
         {
+            _disposed = true;

```

Comment:
The order of setting this flag and calling StopAsync() will affect behavior in the EventProcessor.ProcessEventsAsync().  Consider the following scenario:
1. Host is shutting down and calls this Dispose. 
2. _disposed is true
3. StopAsync() is called which blocks until all in-flight executions complete
4. The host is not yet disposed as this method has not returned and so executions will reach customer code as TryExecuteAsync() will not throw an exception due to a disposed host.  If there is a genuine failed execution in customer code, we will end up not advancing the checkpoint due to logic in line 232.

Recommend setting _disposed = true after the call to StopAsync() completes.  I don't believe we will see any function execution exception due to disposed host as long as the JobHost waits for this Dispose call to complete.

Reply:
if `StopAsync()` will block host shutting down, do we still need _diposed at all?

Reply:
We decided to rely on calling StopAsync() safely via a semaphore _stopped flag so no need for a _disposed flag.  We don't expect any more function triggers after StopAsync() unregisters the handler so any failed in-flight executions should continue to advance the checkpoint.